tests/common/fixtures/advanced_reboot.py

```diff
@@ -639,6 +682,33 @@ def __restorePrevImage(self):
                 wait = self.readyTimeout
             )
 
+    def __restorePrevDockerImage(self):
+        """Restore previous docker image
+        """
+        for service_name, data in self.service_data.items():
+            if data['image_path_on_dut'] is None:
+                self.duthost.shell('sudo config warm_restart disable {}'.format(service_name))
+                continue
+
+            self.duthost.shell('rm -f {}'.format(data['image_path_on_dut']))
+            logger.info('Restore docker image for {}'.format(service_name))
+            self.duthost.shell('service {} stop'.format(service_name))
+            self.duthost.shell('docker rm {}'.format(service_name))
+            image_ids = self.duthost.shell('docker images {} --format \{{\{{.ID\}}\}}'.format(data['image_name']))['stdout_lines']
+            for image_id in image_ids:
+                if image_id != data['image_id']:
+                    self.duthost.shell('docker rmi -f {}'.format(image_id))
+                    break
+
+            self.duthost.shell('docker tag {} {}:{}'.format(data['image_id'], data['image_name'], data['image_tag']))
```

Comment:
Why not "sonic_installer rollback-docker" ?

Reply:
I added comment here:
```
We don't use sonic-installer rollback-docker CLI here because:
1. it does not remove the docker image which causes the running container still using the old image. In that case, we still need do the same thing as we do here.
2. it runs service restart for some containers which enlarge the test duration
```

Reply:
Maybe we need to make "rollback-docker" more useful for the user, so that he don't need to perform many additional steps?

Reply:
Will check.